internal/reminder/reminder.go

```diff
@@ -74,21 +78,52 @@ func (r *reminder) Start(ctx context.Context) error {
 		return errors.New("reminder stopped, cannot start again")
 	default:
 	}
+	defer r.Stop()
```

Comment:
This is kind of odd, since `Start` doesn't return, and would normally be exited by calling `Stop()`.  Add a comment here on why this is needed?

(I'm not saying it's wrong, I'm saying to drop a hint to prevent future authors from deleting the line.

Reply:
This defer here is mostly to handle context cancellation part. Depending on the invoker to clean up reminder after context cancellation isn't something I wanted to do. I added a comment on top.